src/main.cpp

```diff
@@ -204,20 +204,24 @@ void publishMsgs(um6::Registers& r, ros::NodeHandle* imu_nh, sensor_msgs::Imu& i
     imu_msg.orientation_covariance[7] = r.covariance.get_scaled(14);
     imu_msg.orientation_covariance[8] = r.covariance.get_scaled(15);
 
-    // NED -> ENU conversion (x = y, y = x, z = -z)
+    // body-fixed frame NED to ENU: (x y z)->(x -y -z) or (w x y z)->(x -y -z w)
+    // world frame      NED to ENU: (x y z)->(y  x -z) or (w x y z)->(y  x -z w)
     if (tf_ned_to_enu)
     {
-      imu_msg.orientation.x = r.quat.get_scaled(2);
-      imu_msg.orientation.y = r.quat.get_scaled(1);
-      imu_msg.orientation.z = -r.quat.get_scaled(3);
-      imu_msg.orientation.w = r.quat.get_scaled(0);
-
-      imu_msg.angular_velocity.x = r.gyro.get_scaled(1);
-      imu_msg.angular_velocity.y = r.gyro.get_scaled(0);
+      // world frame
+      imu_msg.orientation.w =  r.quat.get_scaled(2);
+      imu_msg.orientation.x =  r.quat.get_scaled(1);
+      imu_msg.orientation.y = -r.quat.get_scaled(3);
+      imu_msg.orientation.z =  r.quat.get_scaled(0);
+
+      // body-fixed frame
+      imu_msg.angular_velocity.x =  r.gyro.get_scaled(0);
+      imu_msg.angular_velocity.y = -r.gyro.get_scaled(1);
       imu_msg.angular_velocity.z = -r.gyro.get_scaled(2);
 
-      imu_msg.linear_acceleration.x = r.accel.get_scaled(1);
-      imu_msg.linear_acceleration.y = r.accel.get_scaled(0);
+      // body-fixed frame
+      imu_msg.linear_acceleration.x =  r.accel.get_scaled(0);
+      imu_msg.linear_acceleration.y = -r.accel.get_scaled(1);
```

Comment:
Issue with customers' Husky is reading zero yaw when pointing to North. The Husky uses a `imu_data_transformer` from `imu_pipeline` to convert from NED to ENU [here](https://github.com/husky/husky_robot/blob/indigo-devel/husky_bringup/launch/um6_config/um6.launch#L8) rather than using it from here.  According to the [datasheet](https://www.chrobotics.com/docs/UM6_datasheet.pdf) on page 16, the angular rates and linear acceleration are given in the body frame.